### DISCUSSION TITLE
How to solve the linear elasticity problems that different patches have different material properties?

### DISCUSSION THREAD

**Discussion by TJUZMX** [gismo/gismo#568] [Q&A] [answered]
Hello

I want to solve the linear elasticity problem, how can I realize that different patches have different material properties, such as elasticity modulus, poisson ratio? Is there any code examples available？

Best regards
Tony

**Comment by hverhelst** [comment, marked as answer]
Dear Tony, this is typically done using the `gsPiecewiseFunction` class. You can make for each patch a gsFunction with the properties etc, and then make a piece-wise function as:
```
gsPiecewiseFunction<> fun(2); // on two patches
fun.addPiece(fun1);
fun.addPiece(fun2);
```
